tests/system_tests/test_core/test_torch_full.py

```diff
@@ -156,55 +156,62 @@ def conv3x3(in_channels, out_channels, **kwargs):
 
 # TODO(jhr): does not work with --flake-finder
 @pytest.mark.xfail(reason="TODO: fix this test")
-def test_all_logging(relay_server, wandb_init):
+def test_all_logging(wandb_backend_spy):
     pytest.importorskip("torch")
     n = 3
-    with relay_server() as relay:
-        run = wandb_init()
-        net = ConvNet()
-        run.watch(
-            net,
-            log="all",
-            log_freq=1,
-        )
-        for _ in range(n):
-            output = net(
-                dummy_torch_tensor(
-                    (32, 1, 28, 28),
-                )
+    run = wandb.init()
+    net = ConvNet()
+    run.watch(
+        net,
+        log="all",
+        log_freq=1,
+    )
+    for _ in range(n):
+        output = net(
+            dummy_torch_tensor(
+                (32, 1, 28, 28),
             )
-            grads = torch.ones(32, 10)
-            output.backward(grads)
-            run.log({"a": 2})
-        run.finish()
-
-    history = relay.context.get_run_history(run.id, include_private=True)
-    assert history.shape == (n, 21)  # it's 21 instead of 20 since we add __run_id
-    assert history["_step"].to_list() == [i for i in range(3)]
-    for i in range(n):
-        assert len(history["parameters/fc2.bias"][i]["bins"]) == 65
-        assert len(history["gradients/fc2.bias"][i]["bins"]) == 65
-
-
-def test_embedding_dict_watch(relay_server, wandb_init):
+        )
+        grads = torch.ones(32, 10)
+        output.backward(grads)
+        run.log({"a": 2})
+    run.finish()
+
+    with wandb_backend_spy.freeze() as snapshot:
+        history = snapshot.history(run_id=run.id)
+        # assert history.shape == (n, 21)  # it's 21 instead of 20 since we add __run_id
```

Comment:
```suggestion
```

Reply:
wait, on second thought why do comment out these lines?

Reply:
These lines were commented out because I was using them as a reference as I translated the test.

It is slightly different because we dont use pandas dataframes, and there was some extra logic appending the `__run_id` to the history we do not do with the backend spy.